pymc3/smc/smc.py

```diff
@@ -184,109 +245,180 @@ def update_weights_beta(self):
             log_weights_un = (new_beta - old_beta) * self.likelihood_logp
             log_weights = log_weights_un - logsumexp(log_weights_un)
 
-        self.log_marginal_likelihood += logsumexp(log_weights_un) - np.log(self.draws)
         self.beta = new_beta
         self.weights = np.exp(log_weights)
         # We normalize again to correct for small numerical errors that might build up
         self.weights /= self.weights.sum()
+        self.log_marginal_likelihood += logsumexp(log_weights_un) - np.log(self.draws)
 
     def resample(self):
-        """Resample particles based on importance weights."""
+        """Resample particles based on importance weights"""
         resampling_indexes = np.random.choice(
             np.arange(self.draws), size=self.draws, p=self.weights
         )
 
-        self.posterior = self.posterior[resampling_indexes]
+        self.population = self.population[resampling_indexes]
         self.prior_logp = self.prior_logp[resampling_indexes]
         self.likelihood_logp = self.likelihood_logp[resampling_indexes]
-        self.posterior_logp = self.prior_logp + self.likelihood_logp * self.beta
-        if self.save_sim_data:
-            self.sim_data = self.sim_data[resampling_indexes]
+        self.tempered_posterior_logp = self.prior_logp + self.likelihood_logp * self.beta
```

Comment:
do we need to clarify that this is the tempered posterior? If that's the case then my previous comments should be to use `self.tempered_posterior` instead of `self.population`.

Reply:
We don't need to... let's discuss on the meeting :)